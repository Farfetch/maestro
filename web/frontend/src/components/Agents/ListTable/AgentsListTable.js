import { DisconnectOutlined, FileTextOutlined } from "@ant-design/icons";
import { Button, Col, message, Row, Space, Switch, Table } from "antd";
import PropTypes from "prop-types";
import React, { useState } from "react";
import { Link } from "react-router-dom";

import { updateAgent } from "../../../lib/api/endpoints/agent";
import { agentStatus as agentStatusModel } from "../../../lib/api/models";
import { agentLogsUrl } from "../../../lib/routes";
import AgentStatusBadge from "../../badge/AgentStatusBadge";

const agentMapper = ({
  id,
  agentStatus,
  hostname,
  ip,
  createdAt,
  updatedAt
}) => ({
  key: id,
  agentStatus,
  hostname,
  ip,
  createdAt,
  updatedAt
});

const AgentsListTable = ({ agents, isLoading, updateTestPlans }) => {
  const [showDisabledAgents, setShowDisabledAgents] = useState(false);
  const agentsDataSource = agents
    .map(agentMapper)
    .filter((agent) => showDisabledAgents || agent.agentStatus !== "DISABLED");

  const disableAgent = async (agentId) => {
    await updateAgent(agentId, {
      agent_status: "DISABLED"
    });
    message.success({ content: "Agent Disabled", duration: 5 });
    updateTestPlans();
  };

  const columns = [
    {
      title: "Hostname",
      dataIndex: "hostname",
      key: "hostname",
      width: 400,
      ellipsis: true,
      sorter: {
        compare: (recordA, recordB) =>
          recordB.hostname.localeCompare(recordA.hostname)
      },
      defaultSortOrder: "descend"
    },
    {
      title: "IP",
      dataIndex: "ip",
      key: "ip",
      width: 150
    },
    {
      title: "Status",
      dataIndex: "agentStatus",
      key: "agentStatus",
      width: 180,
      filters: Object.values(agentStatusModel).map((agentStatus) => ({
        text: agentStatus,
        value: agentStatus
      })),
      filterSearch: true,
      onFilter: (value, record) => record.agentStatus === value,
      render: (text, record) => (
        <AgentStatusBadge
          agentStatus={record.agentStatus}
          text={record.agentStatus}
        />
      )
    },
    {
      title: "Created at",
      dataIndex: "createdAt",
      key: "createdAt",
      width: 180,
      sorter: {
        compare: (recordA, recordB) => recordB.createdAt.diff(recordA.createdAt)
      },
      render: (text, record) => record.createdAt.format("L HH:mm:ss")
    },
    {
      title: "Updated at",
      dataIndex: "updatedAt",
      key: "updatedAt",
      width: 180,
      sorter: {
        compare: (recordA, recordB) => recordB.updatedAt.diff(recordA.updatedAt)
      },
      render: (text, record) => record.updatedAt.format("L HH:mm:ss")
    },
    {
      key: "action",
      width: 100,
      render: (text, record) => (
        <Space size="small">
          <Button
            type="link"
            icon={<DisconnectOutlined />}
            onClick={() => {
              disableAgent(record.key);
            }}
            disabled={
              record.agentStatus === "DISABLED" ||
              record.agentStatus === "AVAILABLE" ||
              record.agentStatus === "RUNNING_TEST"
            }
          >
            Disable
          </Button>
          <Link to={agentLogsUrl(record.key)}>
            <Button icon={<FileTextOutlined />}></Button>
          </Link>{" "}
        </Space>
      )
    }
  ];

  const handleDisabledAgents = (value) => {
    setShowDisabledAgents(value);
  };

  return (
    <Row justify="end" align="middle">
      <Col
        span={12}
        style={{
          display: "flex",
          justifyContent: "flex-end",
          alignItems: "center",
          paddingBottom: "10px"
        }}
      >
        <span>Show Disabled Agents</span>
        <Switch
          onChange={(value) => {
            handleDisabledAgents(value);
          }}
          style={{ marginLeft: 8 }}
        />
      </Col>
      <Col span={24}>
        <Table
          size="small"
          isLoading={isLoading}
          dataSource={agentsDataSource}
          columns={columns}
          pagination={false}
        />
      </Col>
    </Row>
  );
};

AgentsListTable.propTypes = {
  runConfigurations: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      agentStatus: PropTypes.oneOf(Object.values(agentStatusModel)).isRequired,
      hostname: PropTypes.string.isRequired,
      ip: PropTypes.string.isRequired,
      createdAt: PropTypes.object.isRequired,
      updatedAt: PropTypes.object.isRequired
    })
  ),
  isLoading: PropTypes.bool.isRequired
};

export default AgentsListTable;
