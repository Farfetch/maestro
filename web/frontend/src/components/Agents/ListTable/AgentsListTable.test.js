import { render } from "@testing-library/react";
import moment from "moment";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import AgentsListTable from "./AgentsListTable";

describe("components/Agents/ListTable", () => {
  const agents = [
    {
      id: "agent-id",
      ip: "agent IP",
      hostname: "agent hostname",
      agentStatus: "CREATING",
      createdAt: moment("2018-05-24T13:48:04.313000"),
      updatedAt: moment("2018-05-24T13:48:04.313000")
    },
    {
      id: "agent-id",
      ip: "agent IP",
      hostname: "agent hostname",
      agentStatus: "DISABLED",
      createdAt: moment("2019-05-24T13:48:04.313000"),
      updatedAt: moment("2019-05-24T13:48:04.313000")
    }
  ];
  test(`should render AgentsListTable component`, async () => {
    const rendered = render(
      <MemoryRouter initialEntries={["/agents"]}>
        <AgentsListTable agents={agents} isLoading={false} />
      </MemoryRouter>
    );

    const badgeComponent = rendered.container;
    expect(badgeComponent.outerHTML).toMatchSnapshot();
  });

  test(`should render AgentsListTable loader`, async () => {
    const rendered = render(
      <MemoryRouter initialEntries={["/agents"]}>
        <AgentsListTable agents={agents} isLoading={true} />
      </MemoryRouter>
    );

    const badgeComponent = rendered.container;
    expect(badgeComponent.outerHTML).toMatchSnapshot();
  });
});
