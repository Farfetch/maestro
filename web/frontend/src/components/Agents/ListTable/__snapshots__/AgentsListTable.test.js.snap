// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Agents/ListTable should render AgentsListTable component 1`] = `"<div><div class=\\"ant-row ant-row-end ant-row-middle\\"><div style=\\"display: flex; justify-content: flex-end; align-items: center; padding-bottom: 10px;\\" class=\\"ant-col ant-col-12\\"><span>Show Disabled Agents</span><button style=\\"margin-left: 8px;\\" type=\\"button\\" role=\\"switch\\" aria-checked=\\"false\\" class=\\"ant-switch\\"><div class=\\"ant-switch-handle\\"></div><span class=\\"ant-switch-inner\\"></span></button></div><div class=\\"ant-col ant-col-24\\"><div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table ant-table-small\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: fixed;\\"><colgroup><col style=\\"width: 400px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 160px;\\"><col style=\\"width: 160px;\\"><col style=\\"width: 160px;\\"><col style=\\"width: 160px;\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th aria-sort=\\"descending\\" class=\\"ant-table-cell ant-table-column-sort ant-table-cell-ellipsis ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Hostname</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down active\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\">IP</th><th class=\\"ant-table-cell\\"><div class=\\"ant-table-filter-column\\"><span class=\\"ant-table-column-title\\">Status</span><span role=\\"button\\" tabindex=\\"-1\\" class=\\"ant-dropdown-trigger ant-table-filter-trigger\\"><span role=\\"img\\" aria-label=\\"filter\\" class=\\"anticon anticon-filter\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"filter\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z\\"></path></svg></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Created at</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Updated at</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\"></th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"agent-id\\" class=\\"ant-table-row ant-table-row-level-0\\"><td title=\\"agent hostname\\" class=\\"ant-table-cell ant-table-column-sort ant-table-cell-ellipsis\\">agent hostname</td><td class=\\"ant-table-cell\\">agent IP</td><td class=\\"ant-table-cell\\"><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-warning\\"></span><span class=\\"ant-badge-status-text\\">CREATING</span></span></td><td class=\\"ant-table-cell\\">05/24/2018 13:48:04</td><td class=\\"ant-table-cell\\">05/24/2018 13:48:04</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-link\\"><span role=\\"img\\" aria-label=\\"disconnect\\" class=\\"anticon anticon-disconnect\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"disconnect\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M832.6 191.4c-84.6-84.6-221.5-84.6-306 0l-96.9 96.9 51 51 96.9-96.9c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204l-96.9 96.9 51.1 51.1 96.9-96.9c84.4-84.6 84.4-221.5-.1-306.1zM446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l96.9-96.9-51.1-51.1-96.9 96.9c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l96.9-96.9-51-51-96.8 97zM260.3 209.4a8.03 8.03 0 00-11.3 0L209.4 249a8.03 8.03 0 000 11.3l554.4 554.4c3.1 3.1 8.2 3.1 11.3 0l39.6-39.6c3.1-3.1 3.1-8.2 0-11.3L260.3 209.4z\\"></path></svg></span><span>Disable</span></button></div><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><a href=\\"/agent/agent-id/logs\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-default ant-btn-icon-only\\"><span role=\\"img\\" aria-label=\\"file-text\\" class=\\"anticon anticon-file-text\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"file-text\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM504 618H320c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM312 490v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H320c-4.4 0-8 3.6-8 8z\\"></path></svg></span></button></a></div><div class=\\"ant-space-item\\"> </div></div></td></tr></tbody></table></div></div></div></div></div></div></div></div></div>"`;

exports[`components/Agents/ListTable should render AgentsListTable loader 1`] = `"<div><div class=\\"ant-row ant-row-end ant-row-middle\\"><div style=\\"display: flex; justify-content: flex-end; align-items: center; padding-bottom: 10px;\\" class=\\"ant-col ant-col-12\\"><span>Show Disabled Agents</span><button style=\\"margin-left: 8px;\\" type=\\"button\\" role=\\"switch\\" aria-checked=\\"false\\" class=\\"ant-switch\\"><div class=\\"ant-switch-handle\\"></div><span class=\\"ant-switch-inner\\"></span></button></div><div class=\\"ant-col ant-col-24\\"><div class=\\"ant-table-wrapper\\"><div class=\\"ant-spin-nested-loading\\"><div class=\\"ant-spin-container\\"><div class=\\"ant-table ant-table-small\\"><div class=\\"ant-table-container\\"><div class=\\"ant-table-content\\"><table style=\\"table-layout: fixed;\\"><colgroup><col style=\\"width: 400px;\\"><col style=\\"width: 100px;\\"><col style=\\"width: 160px;\\"><col style=\\"width: 160px;\\"><col style=\\"width: 160px;\\"><col style=\\"width: 160px;\\"></colgroup><thead class=\\"ant-table-thead\\"><tr><th aria-sort=\\"descending\\" class=\\"ant-table-cell ant-table-column-sort ant-table-cell-ellipsis ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Hostname</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down active\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\">IP</th><th class=\\"ant-table-cell\\"><div class=\\"ant-table-filter-column\\"><span class=\\"ant-table-column-title\\">Status</span><span role=\\"button\\" tabindex=\\"-1\\" class=\\"ant-dropdown-trigger ant-table-filter-trigger\\"><span role=\\"img\\" aria-label=\\"filter\\" class=\\"anticon anticon-filter\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"filter\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M349 838c0 17.7 14.2 32 31.8 32h262.4c17.6 0 31.8-14.3 31.8-32V642H349v196zm531.1-684H143.9c-24.5 0-39.8 26.7-27.5 48l221.3 376h348.8l221.3-376c12.1-21.3-3.2-48-27.7-48z\\"></path></svg></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Created at</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell ant-table-column-has-sorters\\"><div class=\\"ant-table-column-sorters\\"><span class=\\"ant-table-column-title\\">Updated at</span><span class=\\"ant-table-column-sorter ant-table-column-sorter-full\\"><span class=\\"ant-table-column-sorter-inner\\"><span role=\\"img\\" aria-label=\\"caret-up\\" class=\\"anticon anticon-caret-up ant-table-column-sorter-up\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-up\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M858.9 689L530.5 308.2c-9.4-10.9-27.5-10.9-37 0L165.1 689c-12.2 14.2-1.2 35 18.5 35h656.8c19.7 0 30.7-20.8 18.5-35z\\"></path></svg></span><span role=\\"img\\" aria-label=\\"caret-down\\" class=\\"anticon anticon-caret-down ant-table-column-sorter-down\\"><svg viewBox=\\"0 0 1024 1024\\" focusable=\\"false\\" data-icon=\\"caret-down\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z\\"></path></svg></span></span></span></div></th><th class=\\"ant-table-cell\\"></th></tr></thead><tbody class=\\"ant-table-tbody\\"><tr data-row-key=\\"agent-id\\" class=\\"ant-table-row ant-table-row-level-0\\"><td title=\\"agent hostname\\" class=\\"ant-table-cell ant-table-column-sort ant-table-cell-ellipsis\\">agent hostname</td><td class=\\"ant-table-cell\\">agent IP</td><td class=\\"ant-table-cell\\"><span class=\\"ant-badge ant-badge-status ant-badge-not-a-wrapper\\"><span class=\\"ant-badge-status-dot ant-badge-status-warning\\"></span><span class=\\"ant-badge-status-text\\">CREATING</span></span></td><td class=\\"ant-table-cell\\">05/24/2018 13:48:04</td><td class=\\"ant-table-cell\\">05/24/2018 13:48:04</td><td class=\\"ant-table-cell\\"><div class=\\"ant-space ant-space-horizontal ant-space-align-center\\"><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-link\\"><span role=\\"img\\" aria-label=\\"disconnect\\" class=\\"anticon anticon-disconnect\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"disconnect\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M832.6 191.4c-84.6-84.6-221.5-84.6-306 0l-96.9 96.9 51 51 96.9-96.9c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204l-96.9 96.9 51.1 51.1 96.9-96.9c84.4-84.6 84.4-221.5-.1-306.1zM446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l96.9-96.9-51.1-51.1-96.9 96.9c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l96.9-96.9-51-51-96.8 97zM260.3 209.4a8.03 8.03 0 00-11.3 0L209.4 249a8.03 8.03 0 000 11.3l554.4 554.4c3.1 3.1 8.2 3.1 11.3 0l39.6-39.6c3.1-3.1 3.1-8.2 0-11.3L260.3 209.4z\\"></path></svg></span><span>Disable</span></button></div><div class=\\"ant-space-item\\" style=\\"margin-right: 8px;\\"><a href=\\"/agent/agent-id/logs\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-default ant-btn-icon-only\\"><span role=\\"img\\" aria-label=\\"file-text\\" class=\\"anticon anticon-file-text\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"file-text\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M854.6 288.6L639.4 73.4c-6-6-14.1-9.4-22.6-9.4H192c-17.7 0-32 14.3-32 32v832c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V311.3c0-8.5-3.4-16.7-9.4-22.7zM790.2 326H602V137.8L790.2 326zm1.8 562H232V136h302v216a42 42 0 0042 42h216v494zM504 618H320c-4.4 0-8 3.6-8 8v48c0 4.4 3.6 8 8 8h184c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8zM312 490v48c0 4.4 3.6 8 8 8h384c4.4 0 8-3.6 8-8v-48c0-4.4-3.6-8-8-8H320c-4.4 0-8 3.6-8 8z\\"></path></svg></span></button></a></div><div class=\\"ant-space-item\\"> </div></div></td></tr></tbody></table></div></div></div></div></div></div></div></div></div>"`;
